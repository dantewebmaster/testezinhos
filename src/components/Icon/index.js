import React, { useState } from 'react';
import styled from 'styled-components';
import fa from '../../assets/fa-regular-400.woff';

const Container = styled.i`
  @font-face {
    font-family: "fa_font";
    src: url(${fa});
  }

  font-family: 'fa_font';
  font-size: 30px;

  &:before {
    content: "\\${props => props.unicode}";
    color: ${props => props.color};
    font-style: normal;
  }
`;

export default function Icon({ unicode }) {
  const [iconColor, setIconColor] = useState("#fff");
  const [clickCount, setClickCount] = useState(0);

  function clickHandler() {
    console.log("funfouuuuuu!");

    setClickCount(clickCount + 1);

    if (clickCount >= 3 && clickCount <= 10) {
      setIconColor("#ddd");
    } else if (clickCount >= 10) {
      setIconColor("#000");
    }
  }

  return (
    <>
      <Container 
        color={iconColor} 
        unicode={unicode} 
        onClick={clickHandler} 
      />
      <p>{clickCount}</p>
    </>
  )
}
