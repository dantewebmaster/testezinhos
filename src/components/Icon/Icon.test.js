import React from 'react';
import { shallow, mount } from 'enzyme';
import { create } from 'react-test-renderer';
import 'jest-styled-components';
import Icon from '.';

describe('Home page init', () => {
  it('must renders welcome icon', () => {
    const icon = create(<Icon />);

    expect(icon).toMatchSnapshot()
  });

  it('onclick', () => {
    const icon = shallow(<Icon />);
    icon.simulate('click');
  })
});
