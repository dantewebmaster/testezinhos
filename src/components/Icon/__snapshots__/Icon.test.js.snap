// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page init must renders welcome icon 1`] = `
Array [
  .c0 {
  font-family: 'fa_font';
  font-size: 30px;
}

.c0:before {
  content: "\\";
  color: #fff;
  font-style: normal;
}

<i
    className="c0"
    color="#fff"
    onClick={[Function]}
  />,
  <p>
    0
  </p>,
]
`;
